docs/en/query_language/functions/array_functions.md

```diff
@@ -171,6 +171,47 @@ hasAny(array1, array2)
 
 `SELECT hasAll([[1, 2], [3, 4]], [[1, 2], [1, 2]])` returns `1`.
 
+## hasSubSeq {#hassubseq}
+
+Checks whether one array is a subset of another.
+
+``` sql
+hasAll(sequence, subsequence)
+```
+
+**Parameters**
+
+-   `sequence` – Array of any type with a set of elements.
+-   `subsequence` – Array of any type with elements that should be tested to be a subsequence of `sequence`.
+
+**Return values**
+
+-   `1`, if `sequence` contains `subsequence`.
+-   `0`, otherwise.
+
+**Peculiar properties**
+
+-   An empty array is a subsequence of any array.
+-   `Null` processed as a value.
+-   Order of values in both of arrays does matter.
```

Comment:
@r-zenine i think it worths to mention the algorithm is an o(m*n) implementation and not a KMP. for small m and n, it probably doesn't make big difference but for huge sequence KMP should be better.

Reply:
@huiwang, I agree. Howerver,  the rest of the documentation does not go into that level of  detail for `array` functions so I suggest we follow their current convention.

Reply:
Also, It is not a string, but arrays of generic type. Are you aware of faster search implementation for that case. It would actually help a lot.